state/tx/indexer/kv.go

```diff
@@ -0,0 +1,55 @@
+package indexer
+
+import (
+	"bytes"
+	"fmt"
+
+	db "github.com/tendermint/go-db"
+	"github.com/tendermint/go-wire"
+	"github.com/tendermint/tendermint/types"
+)
+
+// KV is a simpliest possible indexer, backed by Key-Value storage (levelDB).
+// It could only index transaction by its identifier.
+type KV struct {
+	store db.DB
+}
+
+// NewKV returns new instance of KV indexer.
+func NewKV(store db.DB) *KV {
+	return &KV{store: store}
+}
+
+// Tx gets transaction from the KV storage and returns it or nil if the
+// transaction is not found.
+func (indexer *KV) Tx(hash string) (*types.TxResult, error) {
+	if hash == "" {
+		return nil, ErrorEmptyHash
+	}
+
+	rawBytes := indexer.store.Get([]byte(hash))
+	if rawBytes == nil {
+		return nil, nil
```

Comment:
I think `{nil, nil}` tuple is self-describing. Plus I was trying to copy bleve interface, so later we could allow to use external indexers (https://github.com/blevesearch/bleve/blob/master/index.go#L165)